landing-page/scripts/ic_api.py

```
@@ -331,6 +331,37 @@ def ic_manage_url(project_id: int, user=Depends(get_current_user)) -> dict:
+    with db_cursor() as (con, cur):
+        require_project_owner(cur, project_id, user["id"])
+    try:
+        # Short timeout on purpose: this runs on every project-page open, and
+        # a down IC must not park a request thread on the default 30s.
+        _, raw = _get(f"{IC_API_URL}/sessions?project_id={project_id}", timeout=5)
+    except urllib.error.URLError as exc:
+        raise _ic_unreachable(exc)
+    try:
+        sessions = json.loads(raw)
+    except json.JSONDecodeError:
+        raise HTTPException(status_code=502, detail="IC returned a malformed session list")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)